com.unity.render-pipelines.high-definition/Editor/RenderPipeline/ScalableSettingLevelParameterEditor.cs

```diff
@@ -22,13 +22,14 @@ public override bool OnGUI(SerializedDataParameter parameter, GUIContent title)
             rect.y += 2;
             rect.width -= 3;
 
-            o.levelAndOverride = SerializedScalableSettingValueUI.LevelFieldGUI(
+            var levelAndOverride = SerializedScalableSettingValueUI.LevelFieldGUI(
                 rect,
                 title,
                 ScalableSettingSchema.GetSchemaOrNull(ScalableSettingSchemaId.With3Levels),
                 level,
                 useOverride
             );
+            value.intValue = levelAndOverride.level;
```

Comment:
I don't think this works.

If the user overrides the value the level returned is `-1` and if you store this value the behaviour will be undefined.

While reading the volume serialization framework, I see that the `GetObjectRef` return a C# reference of an object, and I don't know if any modification of this object will be applied to the `SerializedProperty`.

I suggest to check how to apply the modifications done to the SerializedDataParameter` to the underlying `SerializedProperty`

Reply:
> While reading the volume serialization framework, I see that the GetObjectRef return a C# reference of an object, and I don't know if any modification of this object will be applied to the SerializedProperty.
It doesn't. That's why we had issues with undo/redo and scene refresh.
> f the user overrides the value the level returned is -1 and if you store this value the behaviour will be undefined
I'm not sure I'm following you. I tried override and it worked just the same as far as I can tell.
From what I see, all other parameters work the same way. See FloatParameterDrawer for example.

Reply:
Assigning only the level and ignoring the `useOverride` sounds like an issue.

IMO, you should instead encode the tuple into an int in the same way `ScalableSettingLevelParameter.levelAndOverride.set` does. (like creating a static function for this).